### PR TITLE
josm: 17702 → 17833

### PR DESCRIPTION
**Pull request by sikmir**
###### Motivation for this change
* [2021-04-28: Stable release 17833](https://josm.openstreetmap.de/wiki/Changelog#stable-release-21.04)
* make it easy to override Java options:
```
(josm.overrideAttrs (old: {
  extraJavaOpts = "-Xmx2048m";
}))
```

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121038` at 8afd4951 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>josm</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/misc/josm/default.nix:29:3:
```
   |
29 |   installPhase =
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/misc/josm/default.nix [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
+stdenv.mkDerivation rec {
   inherit pname version;
 
   dontUnpack = true;
 
   nativeBuildInputs = [ makeWrapper ];
   buildInputs = lib.optionals (!stdenv.isDarwin) [ jre ];
 
+  extraJavaOpts = "-Djosm.restart=true -Djava.net.useSystemProxies=true";
+
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/josm/default.nix:30]:
```suggestion
```

File: pkgs/applications/misc/josm/default.nix [hunk 2]
```diff
@@ -1,31 +1,33 @@
 { lib, stdenv, fetchurl, fetchsvn, makeWrapper, unzip, jre, libXxf86vm }:
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/josm/default.nix:1]:
```suggestion
{ lib, stdenv, fetchurl, fetchsvn, makeWrapper, unzip, jre, libXxf86vm
, extraJavaOpts ? "-Djosm.restart=true -Djava.net.useSystemProxies=true"
}:
```

**Comment by SuperSandro2000** [comment]
Using inputs we can override it with override which is most of the time a bit easier.

**Pull request merged**